spark/src/main/java/org/opensearch/sql/spark/config/SparkExecutionEngineConfigSupplierImpl.java

```diff
@@ -0,0 +1,42 @@
+package org.opensearch.sql.spark.config;
+
+import static org.opensearch.sql.common.setting.Settings.Key.CLUSTER_NAME;
+import static org.opensearch.sql.common.setting.Settings.Key.SPARK_EXECUTION_ENGINE_CONFIG;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import lombok.AllArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.opensearch.cluster.ClusterName;
+import org.opensearch.sql.common.setting.Settings;
+
+@AllArgsConstructor
+public class SparkExecutionEngineConfigSupplierImpl implements SparkExecutionEngineConfigSupplier {
+
+  private Settings settings;
+
+  @Override
+  public SparkExecutionEngineConfig getSparkExecutionEngineConfig() {
+    String sparkExecutionEngineConfigSettingString =
+        this.settings.getSettingValue(SPARK_EXECUTION_ENGINE_CONFIG);
+    SparkExecutionEngineConfig sparkExecutionEngineConfig = new SparkExecutionEngineConfig();
+    if (!StringUtils.isBlank(sparkExecutionEngineConfigSettingString)) {
+      SparkExecutionEngineConfigClusterSetting sparkExecutionEngineConfigClusterSetting =
```

Comment:
why not read SparkExecutionEngineConfig from json?

Reply:
User can set clustername in the setting and it can be set in execution config unknowingly.
So wanted to have a separate class for each.